src/adc_readout/AdcParse.cpp

```diff
@@ -95,24 +95,50 @@ HeaderInfo parseHeader(const InData &Packet) {
   auto HeaderRaw = reinterpret_cast<const PacketHeader *>(Packet.Data);
   PacketHeader Header(*HeaderRaw);
   Header.fixEndian();
-  if (IDLE_PACKET == Header.PacketType) {
-    ReturnInfo.Type = PacketType::Idle;
-  } else if (DATA_PACKET == Header.PacketType) {
-    ReturnInfo.Type = PacketType::Data;
-  } else {
+  if (Header.Type != PacketType::Idle and Header.Type != PacketType::Data) {
     throw ParserException(ParserException::Type::HEADER_TYPE);
   }
+  ReturnInfo.Type = Header.Type;
   ReturnInfo.DataStart = sizeof(PacketHeader);
   ReturnInfo.ReadoutCount = Header.ReadoutCount;
-  ReturnInfo.ReferenceTimestamp = Header.ReferenceTimeStamp;
+  ReturnInfo.ReferenceTimestamp = {Header.ReferenceTimeStamp,
+                                   TimeStamp::ClockMode(Header.ClockMode)};
+  if (Packet.Length != Header.ReadoutLength + 8u) {
```

Comment:
+ 8u? 

I assume this is some sort of padding, but it is probably better to make it explicit by using 
some well named variable.

Reply:
True, done.